cli/tests/environment-pull.bats

```diff
@@ -288,15 +336,111 @@ function make_incompatible() {
 # due to the current system missing <system> in `option.systems`
 # AND a package that is indeed not able to be built for the current system
 # should show a warning, but otherwise succeed to pull
-@test "pull unsupported environment succeeds with '--add-system' flag but shows warning if unable to build still" {
+@test "pull unsupported environment succeeds with '--force' flag but shows warning if unable to build still" {
   update_dummy_env "owner" "name"
 
-  make_incompatible "owner" "name" yes
+  make_incompatible "owner" "name"
+  add_incompatible_package "owner" "name"
 
-  run "$FLOX_BIN" pull --remote owner/name --add-system
+  run "$FLOX_BIN" pull --remote owner/name --force
   assert_success
   assert_line --partial "Could not build modified environment, build errors need to be resolved manually."
 
   run "$FLOX_BIN" list
   assert_success
 }
+
+# ---------------------------------------------------------------------------- #
+
+# bats test_tags=pull:unsupported-package
+# pulling an environment with a package that is not available for the current platform
+# should fail with an error
+@test "pull environment with package not available for the current platform fails" {
+  update_dummy_env "owner" "name"
+  add_incompatible_package "owner" "name"
+
+  run "$FLOX_BIN" pull --remote owner/name
+
+  assert_failure
+  assert_line --partial "package 'extra' is not available for this system ('$NIX_SYSTEM')"
+}
+
+# bats test_tags=pull:unsupported-package:prompt-fail
+# pulling an environment with a package that is not available for the current platform
+# should prompt to ignore the error and pull the environment anyway.
+# When answering no, an error should be shown and the environment should not be pulled.
+@test "pull environment with package not available for the current platform prompts to abort or ignore -- aborts cleanly" {
```

Comment:
Shorten test names and avoid special characters. `bats` converts these into variable names and I think it truncates them at a certain length.
We've already encountered some "duplicate test name" errors on test with very long names that share a common prefix.

Reply:
shortened

Reply:
Thanks.